modules/aws-bastion-instance/main.tf

```diff
@@ -77,13 +72,13 @@ resource "aws_iam_instance_profile" "bastion" {
   role        = aws_iam_role.bastion.id
 }
 
-resource "aws_launch_configuration" "bastion" {
-  name_prefix                 = "${var.name}-"
-  image_id                    = var.ami_id
-  instance_type               = var.instance_type
-  user_data                   = data.template_file.user_data.rendered
-  enable_monitoring           = var.enable_monitoring
-  associate_public_ip_address = true
+resource "aws_instance" "bastion" {
+  ami           = var.ami_id
```

Comment:
Improvement suggestion - If you don't care about AMI, we can provide the default one (and make `var.ami_id` optional):
```tf
data "aws_ami" "amazon_linux" {
  filter {
    name   = "name"
    values = ["amzn2-ami-hvm-*-x86_64-gp2"]
  }
  owners      = ["amazon"]
  most_recent = true
}

resource "aws_instance" "bastion" {
  ami           = var.ami_id ? var.ami_id : data.aws_ami.amazon_linux
  # ...
}
```